core/src/display_object/morph_shape.rs

```diff
@@ -98,6 +104,31 @@ impl<'gc> TDisplayObject<'gc> for MorphShape<'gc> {
         // Noop
     }
 
+    fn object2(&self) -> Avm2Value<'gc> {
+        self.0
+            .read()
+            .object
+            .map(Avm2Value::from)
+            .unwrap_or(Avm2Value::Null)
+    }
+
+    fn set_object2(&self, context: &mut UpdateContext<'_, 'gc>, to: Avm2Object<'gc>) {
+        self.0.write(context.gc_context).object = Some(to);
+    }
+
+    /// Construct objects placed on this frame.
+    fn construct_frame(&self, context: &mut UpdateContext<'_, 'gc>) {
+        if context.is_action_script_3() && matches!(self.object2(), Avm2Value::Null) {
+            let class = context.avm2.classes().morphshape;
+            let mut activation = Avm2Activation::from_nothing(context.reborrow());
```

Comment:
Should this be `from_domain`?

Reply:
No, as nothing tries to access the domain. If this was wrong, it would panic when accessing the missing domain